api/v1beta1/upgradetask_types.go

```diff
@@ -19,32 +19,27 @@ package v1beta1
 import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-)
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+	"github.com/microsoft/cluster-api-inplace-upgrader/stub"
+)
 
 // UpgradeTaskSpec defines the desired state of UpgradeTask
 type UpgradeTaskSpec struct {
 	ClusterRef             *corev1.ObjectReference  `json:"clusterRef,omitempty"`
 	ControlPlaneRef        *corev1.ObjectReference  `json:"controlPlaneRef,omitempty"`
 	MachineDeploymentRef   *corev1.ObjectReference  `json:"machineDeploymentRef,omitempty"`
 	MachinesRequireUpgrade []corev1.ObjectReference `json:"machinesRequireUpgrade,omitempty"`
-	NewMachineSpec         *MachineSpec             `json:"newMachineSpec,omitempty"`
-
-	// // Template defines the template of Machine In-place upgrader
-	// Template *MachineUpgraderTemplate `json:"template,omitempty"`
-	Phase UpgradeTaskPhase `json:"phase,omitempty"`
+	NewMachineSpec         stub.MachineSpec         `json:"newMachineSpec,omitempty"`
 }
 
 // UpgradeTaskStatus defines the observed state of UpgradeTask
 type UpgradeTaskStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	Phase UpgradeTaskPhase `json:"phase,omitempty"`
```

Comment:
any field in status shall only be updated by  cr reconciler. But we need set this phase to abort if we need abort a ongoing upgradeTask.

Reply:
`TargetPhase` is added to address this comment.